server/channels/app/migrations.go

```diff
@@ -23,32 +26,34 @@ const remainingSchemaMigrationsKey = "RemainingSchemaMigrations"
 const postPriorityConfigDefaultTrueMigrationKey = "PostPriorityConfigDefaultTrueMigrationComplete"
 
 // This function migrates the default built in roles from code/config to the database.
-func (a *App) DoAdvancedPermissionsMigration() {
-	a.Srv().doAdvancedPermissionsMigration()
+func (a *App) DoAdvancedPermissionsMigration() error {
+	return a.Srv().doAdvancedPermissionsMigration()
 }
 
-func (s *Server) doAdvancedPermissionsMigration() {
+func (s *Server) doAdvancedPermissionsMigration() error {
 	// If the migration is already marked as completed, don't do it again.
+	var nfErr *store.ErrNotFound
 	if _, err := s.Store().System().GetByName(model.AdvancedPermissionsMigrationKey); err == nil {
-		return
+		return nil
+	} else if !errors.As(err, &nfErr) {
+		return fmt.Errorf("could not query migration: %w", err)
 	}
 
 	mlog.Info("Migrating roles to database.")
 	roles := model.MakeDefaultRoles()
 
-	allSucceeded := true
-
+	var multiErr *multierror.Error
 	for _, role := range roles {
 		_, err := s.Store().Role().Save(role)
 		if err == nil {
 			continue
 		}
+		mlog.Info("Couldn't save the role for advanced permissions migration, this can be an expected case", mlog.Err(err))
```

Comment:
Should this be a `warn`?

Reply:
I'm not sure, because I don't know if it's an actionable thing at this point. But we can convert it to `warn`, it wouldn't hurt.